nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix

```diff
@@ -349,6 +380,82 @@ in {
         Windows can unseal the encryption key.
       '';
     };
+
+    windows = mkOption {
+      default = { };
+      description = ''
+        Make Windows bootable from systemd-boot. This option is not necessary when Windows and
+        NixOS use the same EFI System Partition (ESP). In that case, Windows will automatically be
+        detected by systemd-boot.
+
+        However, if Windows is installed on a separate drive or ESP, you can use this option to add
+        a menu entry for each installation manually.
+
+        The attribute name is the title of the menu entry.
+      '';
+      example = literalExpression ''
+        {
+          "Windows 10".efiDeviceHandle = "HD0c3";
+          "Windows 11... Are you sure about this?" = {
+            efiDeviceHandle = "FS1";
+            sortKey = "z_windows";
+            cleanName = "windows-11";
+          };
+        }
+      '';
+      type = types.attrsOf (types.submodule ({ config, ... }: {
+        options = {
+          efiDeviceHandle = mkOption {
+            type = types.str;
+            example = "HD1b3";
+            description = ''
+              The device handle of the EFI System Partition (ESP) where the Windows bootloader is
+              located. This is the device handle that the EDK2 UEFI Shell uses to load the
+              bootloader.
+
+              To find this handle, follow these steps:
+              1. Set {option}`boot.loader.systemd-boot.edk2-uefi-shell.enable` to `true`
+              2. Run `nixos-rebuild boot`
+              3. Reboot and select "EDK2 UEFI Shell" from the systemd-boot menu
+              4. Run `map -c` to list all consistent device handles
+              5. For each device handle (for example, `HD0c1`), run `ls HD0c1:\EFI`
+              6. If the output contains the directory `Microsoft`, you might have found the correct device handle
+              7. Run `HD0c1:\EFI\Microsoft\Boot\Bootmgfw.efi` to check if Windows boots correctly
+              8. If it does, this device handle is the one you need (in this example, `HD0c1`)
+
+              This option is required, there is no useful default.
+            '';
+          };
+
+          sortKey = mkOption {
+            type = types.str;
+            default = "";
+            defaultText = ''{option}`cleanName`, prefixed with "os_"'';
+            description = ''
+              `systemd-boot` orders the menu entries by their sort keys,
+              so if you want something to appear after all the NixOS entries,
+              it should start with {file}`o` or onwards.
+
+              See also {option}`boot.loader.systemd-boot.sortKey`..
+            '';
+          };
+
+          cleanName = mkOption {
```

Comment:
I feel like the meaning of this should be flipped, i.e.

```nix
{
  "windows-11" = {
    title = "Windows 11";
    efiDeviceHandle = "foo";
  };
}
```

While this means that this isn't a sleek one liner anymore, it would align it more with `boot.loader.systemd-boot.extraEntries`.

Reply:
I went a head and did something slightly nicer:

```
{
  systemd-boot.windows."11".efiDeviceHandle = "foo";
}
```

I feel this is the cleanest solution. The most common usecase is still the simplest one, but the filename becomes an implementation detail while the title can be overridden just like you requested.